core/evalir.ml

```diff
@@ -673,29 +673,34 @@ struct
          | Some (limit, offset) ->
             Some (Value.unbox_int (value env limit), Value.unbox_int (value env offset)) in
        if Settings.get_value Basicsettings.Shredding.shredding then
-         begin
-           match EvalNestedQuery.compile_shredded env (range, e) with
-           | None -> computation env cont e
-           | Some (db, p) ->
-             begin
-               if db#driver_name() <> "postgresql"
-                 then raise (Errors.Runtime_error "Only PostgreSQL database driver supports shredding");
-               let get_fields t =
-                             match t with
-                             | `Record fields ->
-                                 StringMap.to_list (fun name p -> (name, `Primitive p)) fields
-                             | _ -> assert false
-               in
-               let execute_shredded_raw (q, t) =
-                 Database.execute_select_result (get_fields t) q db, t in
-               let raw_results =
-                 EvalNestedQuery.Shred.pmap execute_shredded_raw p in
-               let mapped_results =
-                 EvalNestedQuery.Shred.pmap EvalNestedQuery.Stitch.build_stitch_map raw_results in
-                 apply_cont cont env
-                 (EvalNestedQuery.Stitch.stitch_mapped_query mapped_results)
-             end
-         end
+         match EvalNestedQuery.compile_shredded env (range, e) with
+         | None -> computation env cont e
+         | Some (db, p) ->
+            try
+              let driver = db#driver_name () in
+              if String.iequal_ascii driver "postgresql" || String.iequal_ascii driver "sqlite"
+              then let get_fields t =
+                     match t with
+                     | `Record fields ->
+                        StringMap.to_list (fun name p -> (name, `Primitive p)) fields
+                     | _ -> assert false
+                   in
+                   let execute_shredded_raw (q, t) =
+                     Database.execute_select_result (get_fields t) q db, t in
+                   let raw_results =
+                     EvalNestedQuery.Shred.pmap execute_shredded_raw p in
+                   let mapped_results =
+                     EvalNestedQuery.Shred.pmap EvalNestedQuery.Stitch.build_stitch_map raw_results in
+                   apply_cont cont env
+                     (EvalNestedQuery.Stitch.stitch_mapped_query mapped_results)
+              else raise Not_found (* Static exception. *)
```

Comment:
I do not understand the rationale for this control flow.  Why isn't
```
try (if M then N else raise exn) with Exn -> P
```
equivalent to 
```
if M then N else P
```

??

Reply:
Oh yes, it is. I experimented with something more general; I forgot to remove the `try`.

Reply:
OK.  I don't understand how this commit (699929b) improves matters though.  Also, GitHub user "jcheney" is not me.

Reply:
Oh wait, that commit is from 4 hours ago, it looks like you haven't actually changed anything since my review.